Source/NSOrderedSet.m

```diff
@@ -0,0 +1,1912 @@
+			initWithSet: aSet
+			  copyItems: flag]);
+}
+
+// instance methods
+- (instancetype) initWithArray:(NSArray *)other
+{
+  unsigned	count = [other count];
+
+  if (count == 0)
+    {
+      return [self init];
+    }
+  else
+    {
+      GS_BEGINIDBUF(objs, count);
+
+      if ([other isProxy])
+	{
+	  unsigned	i;
+
+	  for (i = 0; i < count; i++)
+	    {
+	      objs[i] = [other objectAtIndex: i];
+	    }
+	}
+      else
+	{
+          [other getObjects: objs];
+	}
+      self = [self initWithObjects: objs count: count];
+      GS_ENDIDBUF();
+      return self;
+    }
+
+  return nil;
+}
+
+- (instancetype) initWithArray:(NSArray *)other copyItems:(BOOL)flag
+{
+  unsigned	count = [other count];
+  unsigned      j = count;
+  
+  if (count == 0)
+    {
+      return [self init];
+    }
+  
+  GS_BEGINIDBUF(objs, count);
+  {
+    unsigned	i;
+    
+    for (i = 0; i < count; i++)
+      {
+	if (flag == NO)
+	  {
+	    objs[i] = [other objectAtIndex: i];
+	  }
+	else // copy the items.
+	  {
+	    objs[i] = [[other objectAtIndex: i] copy];
+	  }
+      }
+  }
+  
+  self = [self initWithObjects: objs count: count];
+  
+  if(flag == YES)
+    {
+      while(j--)
+	{
+	  [objs[j] release];
+	}
+    }
+  GS_ENDIDBUF();
+  return self;
+}
+
+- (instancetype) initWithArray:(NSArray *)other
+                         range:(NSRange)range
+                     copyItems:(BOOL)flag
+{
+  unsigned	count = [other count];
+  unsigned      i = 0, j = 0;
+    
+  if (count == 0)
+    {
+      return [self init];
+    }
+
+  GS_BEGINIDBUF(objs, range.length);
+  {
+    unsigned      loc = range.location;
+    unsigned      len = range.length;
+    
+    for (i = 0; i < count; i++)
+      {
+	if(i >= loc && j < len)
+	  {
+	    if(flag == YES)
+	      {		  
+		objs[i] = [[other objectAtIndex: i] copy];
+	      }
+	    else
+	      {
+		objs[i] = [other objectAtIndex: i];
+	      }
+	    j++;
+	  }
+	
+	if(j >= len)
+	  {
+	    break;
+	  }
+      }
+  }
+  self = [self initWithObjects: objs count: count];
+
+  if(flag == YES)
+    {
+      while(j--)
+	{
+	  [objs[j] release];
+	}
+    }
+  GS_ENDIDBUF();
+
+  return self;
+}
+
+- (instancetype) initWithObject: (id)obj
+{
+  id objs[] = {obj};
+  
+  self = [self initWithObjects: objs count: 1];
+  if(self == nil)
+    {
+      NSLog(@"Problem initializing with one element");
+    }
+  return self;
+}
+
+- (instancetype) initWithObjects:(id)firstObject, ...
+{
+  GS_USEIDLIST(firstObject,
+    self = [self initWithObjects: __objects count: __count]);
+  return self;
+}
+
+/** <init /> <override-subclass />
+ * Initialize to contain (unique elements of) objects.<br />
+ * Calls -init (which does nothing but maintain MacOS-X compatibility),
+ * and needs to be re-implemented in subclasses in order to have all
+ * other initialisers work.
+ */
+- (instancetype) initWithObjects:(const id [])objects // required override.
+                           count:(NSUInteger)count
+{
+  self = [self init];
```

Comment:
Shouldn't we just raise an exception here?  [self subclassResponsibility: _cmd];